x-pack/plugin/esql/qa/testFixtures/src/main/resources/ip.csv-spec

```diff
@@ -485,3 +485,52 @@ beta         | 127.0.0.1
 beta         | 127.0.0.1
 beta         | 127.0.0.1
 ;
+
+ipPrefix
+//tag::ipPrefix[]
+required_capability: fn_ip_prefix
+row ip4 = to_ip("1.2.3.4"), ip6 = to_ip("fe80::cae2:65ff:fece:feb9")
+| eval ip4_prefix = ip_prefix(ip4, 24), ip4_as_6_prefix = ip_prefix(ip4, 120), ip6_prefix = ip_prefix(ip6, 112), ip6_as_4_prefix = ip_prefix(ip6, 16),
+       ip4_full = ip_prefix(ip4, 32), ip6_full = ip_prefix(ip6, 128),
+       ip4_zeroes = ip_prefix(ip4, 0);
+//end::ipPrefix[]
+
+//tag::ipPrefix-result[]
+ip4:ip  | ip6:ip                    | ip4_prefix:ip | ip4_as_6_prefix:ip | ip6_prefix:ip             | ip6_as_4_prefix:ip        | ip4_full:ip | ip6_full:ip               | ip4_zeroes:ip
+1.2.3.4 | fe80::cae2:65ff:fece:feb9 | 1.2.3.0       | 1.2.3.0            | fe80::cae2:65ff:fece:0000 | fe80::cae2:65ff:fece:0000 | 1.2.3.4     | fe80::cae2:65ff:fece:feb9 | 0.0.0.0
+//end::ipPrefix-result[]
+;
+
+ipPrefixWithBits
+required_capability: fn_ip_prefix
+row ip4 = to_ip("1.2.3.255"), ip6 = to_ip("fe80::cae2:65ff:fece:feff")
+| eval ip4_prefix = ip_prefix(ip4, 25), ip6_prefix = ip_prefix(ip6, 121);
+
+ip4:ip   | ip6:ip                    | ip4_prefix:ip | ip6_prefix:ip
+1.2.3.255 | fe80::cae2:65ff:fece:feff | 1.2.3.128       | fe80::cae2:65ff:fece:fe80
+;
+
+ipPrefixAsGroup
+required_capability: fn_ip_prefix
+from hosts
+| stats count(*) by ip_prefix(ip1, 24);
+warning:Line 2:21: evaluation of [ip_prefix(ip1, 24)] failed, treating result as null. Only first 20 failures recorded.
+warning:Line 2:21: java.lang.IllegalArgumentException: single-value function encountered multi-value
+
+count(*):long | ip_prefix(ip1, 24):ip
+2             | null
+3             | 127.0.0.0
+2             | ::0
+1             | 128.0.0.0
+1             | fe81::cae2:65ff:fece:fe00
+1             | fe80::cae2:65ff:fece:fe00
+;
+
+ipPrefixWithWrongArguments#[skip:-,reason:null returns not supported yet in evaluators]
+required_capability: fn_ip_prefix
+row ip4 = to_ip("1.2.3.4")
+| eval a = ip_prefix(null, 128), b = ip_prefix(ip4, null), c = ip_prefix(ip4, 129), d = ip_prefix(ip4, -1);
```

Comment:
My idea was fixing it after completing this one. If it's merged first, I'd uncomment it. Otherwise, I'd uncomment in the fix.

As it's a global problem, and other functions were _really_ affected (Like `ROW a = substring("", 1, 1)`, which fails because of the "", which returns null), I don't think this PR should be paused (There's no real affectation here imo)

Reply:
Opened issue here: https://github.com/elastic/elasticsearch/issues/109095

Reply:
Thank you!